crates/store/src/db/sql-queries/select_merged_deltas.sql

```diff
@@ -0,0 +1,84 @@
+/*
+ * Selects and merges account deltas by account id and block range.
+ * Note, that `block_start` is exclusive and `block_end` is inclusive.
+ * Parameters:
+ *   ?1: Account ID
+ *   ?2: Block start
+ *   ?3: Block end
+ * Types:
+ *   0: Storage scalar values
+ *   1: Storage map values
+ *   2: Fungible assets
+ *   3: Non-fungible assets
+ */
+
+-- Selects and merges storage deltas by account id and block range (gets latest values by slot).
+SELECT
+    0 AS type, block_num, slot, NULL, value
+FROM
+    account_storage_slot_updates AS a
+WHERE
+    account_id = ?1 AND
+    block_num > ?2 AND
+    block_num <= ?3 AND
+    NOT EXISTS(
+        SELECT 1
+        FROM account_storage_slot_updates AS b
+        WHERE
+            b.account_id = ?1 AND
+            a.slot = b.slot AND
+            a.block_num < b.block_num AND
+            b.block_num <= ?3
+    )
```

Comment:
This is good point, thank you! My approach is more old-school, it forces optimizer to filter by existence, but I think, yours gives it more room for optimizations which might result in better performance. But to answer the question, it will require benchmarking. I put this query to the issue: https://github.com/0xPolygonMiden/miden-node/issues/558